agent/consul/state/intention_test.go

```diff
@@ -1199,6 +1199,73 @@ func TestStore_IntentionsList(t *testing.T) {
 	})
 }
 
+// TestStore_IntentionExact_ConfigEntries test that we can get a local config entry intention
+// and a peered config entry intention with an IntentionGetExact call
+func TestStore_IntentionExact_ConfigEntries(t *testing.T) {
+	s := testConfigStateStore(t)
+	inputs := []*structs.ServiceIntentionsConfigEntry{
+		{
+			Kind: structs.ServiceIntentions,
+			Name: "foo",
+			Sources: []*structs.SourceIntention{
+				{
+					Action:      structs.IntentionActionAllow,
+					Name:        "bar",
+					Peer:        "peer1",
+					Description: "peered service intention",
+				},
+				{
+					Action:      structs.IntentionActionAllow,
+					Name:        "bar",
+					Description: "local service intention",
+				},
+			},
+		},
+	}
+	idx := uint64(0)
+
+	for _, input := range inputs {
+		require.NoError(t, input.Normalize())
+		require.NoError(t, input.Validate())
+		idx++
+		require.NoError(t, s.EnsureConfigEntry(idx, input))
+	}
+
+	// assert that we can get the peered intention
+	testutil.RunStep(t, "assert that we can get the peered intention", func(t *testing.T) {
```

Comment:
These are better off as `t.Run`. It's probably confusing to see different ways of writing subtests, but basically:

`t.Run` should be the default for isolated, independent tests that can pass by themselves when run (goland and VSC are usually able to run well-formed `t.Run` subtests by themselves)

`testutil.RunStep` is used when tests are dependent on the results of a previous subtest (e.g. shared state is mutated in each subtest). These subtests are not independently runnable.

In this case, you have shared state but since all you do is read from it, the two assertions are independent of one another.